src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -1238,31 +1238,33 @@ impl<T: 'static> EventProcessor<T> {
         }
         match self.ime_event_receiver.try_recv() {
             Ok((window, event)) => match event {
+                ImeEvent::Enabled => {
+                    callback(Event::WindowEvent {
+                        window_id: mkwid(window),
+                        event: WindowEvent::IME(IMEEvent::Enabled),
+                    });
+                }
                 ImeEvent::Start => {
                     self.is_composing = true;
                     self.composed_text = None;
                     callback(Event::WindowEvent {
                         window_id: mkwid(window),
-                        event: WindowEvent::Composition(CompositionEvent::CompositionStart(
-                            "".to_owned(),
-                        )),
+                        event: WindowEvent::IME(IMEEvent::Preedit("".to_owned(), None, None)),
                     });
                 }
                 ImeEvent::Update(text, position) => {
                     if self.is_composing {
                         callback(Event::WindowEvent {
                             window_id: mkwid(window),
-                            event: WindowEvent::Composition(CompositionEvent::CompositionUpdate(
-                                text, position,
-                            )),
+                            event: WindowEvent::IME(IMEEvent::Preedit(text, Some(position), None)),
```

Comment:
Also, is there a way to inform that cursor is hidden, thus shouldn't be drawn, if there's such thing we should inform about that via sending `None` and `None`? 

```suggestion
                            event: WindowEvent::IME(IMEEvent::Preedit(text, Some(position), Some(position))),
```

Reply:
Fixed.

In X11, I cannot find any event related to hidden cursor.